project/Boilerplate.scala

```diff
@@ -0,0 +1,82 @@
+import sbt._
+
+import scala.annotation.tailrec
+import scala.collection.immutable
+
+/**
+ * Copied, with some modifications, from https://github.com/milessabin/shapeless/blob/main/project/Boilerplate.scala
+ *
+ * Generate a range of boilerplate classes, those offering alternatives with 0-22 params and would be tedious to craft
+ * by hand
+ *
+ * @author
+ *   Miles Sabin
+ * @author
+ *   Kevin Wright
+ */
+object Boilerplate {
```

Comment:
I believe that should be a separate library to prevent this copy-pasting among cats-* libraries. But there is none at the moment.

Reply:
Like this?

https://github.com/sbt/sbt-boilerplate

Reply:
Yep. I mean that Typelevel could have its own. Because there is an obvious need.